### ISSUE TITLE
Bad primary air system controller index in SolveWaterCoilController

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6413]
Issue overview
--------------
Stumbled across this while addressing #6358. In [SimAirServingZone::SolveWaterCoilcontroller](https://github.com/NREL/EnergyPlus/blame/682bda2e1b9b62fa41965ddd84b23eb5b168035f/src/EnergyPlus/SimAirServingZones.cc#L2944) this line trips an array bounds crash,:
```
			PrimaryAirSystem( AirLoopNum ).ControlConverged( ControllerIndex ) = ControllerConvergedFlag;
```
The problem is that `ControllerIndex` is for the entire array of controllers, but `ControlConverged` is only allocated for the number of controllers on this particular air loop.  This may not be the only place that's using the wrong controller index.

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus 8.9.0-plus #6358  - defect file will fail early without the 6358 fix.  Then it fails here.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
@rraustad This code was added as part of #6197

**Comment by rraustad** [comment]
That's what I get for testing with only a single air loop. We should require any changes be tested using multiple "whatever we are changing".

**Comment by rraustad** [comment]
See PR #6414.

**Comment by mjwitte** [comment]
Closed via #6414
